packages/shopify-api/lib/utils/unified-admin-url-helper.ts

```diff
@@ -0,0 +1,41 @@
+// This class assumes the legacy url has been sanitized already
+// Converts admin.shopify.com/store/my-shop to my-shop.myshopify.com
+export function unifiedAdminUrlToLegacyUrl() {
+  return (unifiedAdminUrl: string): string | null => {
+    const isUnifiedAdminUrl = unifiedAdminUrl.split('.')[0] === 'admin';
+
+    if (!isUnifiedAdminUrl) {
+      return null;
+    }
+
+    const urlComponents = unifiedAdminUrl.split('/');
+    const shopName = urlComponents[urlComponents.length - 1];
+
+    const isSpinUrl = unifiedAdminUrl.includes('spin.dev/store/');
+    if (isSpinUrl) {
+      const spinUrlComponents = unifiedAdminUrl
+        .split('.')
+        .slice(2, 5)
+        .join('.');
```

Comment:
I have no idea how likely these are to change, but this seems slightly brittle to me.

If the URL we get is sanitized, could we use a `new URL` instead of manually parsing the domain? I would imagine that the domain would remain the same, we'd just add `${shopName}.shopify` to the beginning?